core/src/main/java/org/eclipse/hono/util/HonoDeviceNotificationConstants.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 1.0 which is available at
+ * https://www.eclipse.org/legal/epl-v10.html
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ */
+
+package org.eclipse.hono.util;
+
+import org.apache.qpid.proton.message.Message;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * Constants used for device notification events.
+ */
+public final class HonoDeviceNotificationConstants {
+    /**
+     * Prevent construction.
+     */
+    private HonoDeviceNotificationConstants() {
+    }
+
+    /**
+     * The content type that is defined for device notification events.
+     */
+    public static final String CONTENT_TYPE_DEVICE_NOTIFICATION = "application/vnd.eclipse-hono-notification+json";
```

Comment:
the whole idea of using a content type to identify the type of event was based on the assumption, that it is not necessary to introspect (or worse, try to deserialize a payload using Jackson) an event in order to determine if a consumer needs to process it.

Using a generic *notification* type undermines this idea FMPOV. The event for indicating that a device is now ready to receive a command should therefore have its own content-type FMPOV because it is of such importance, will probably be used very frequently and is relevant to almost all consuming applications.

Reply:
> the whole idea of using a content type to identify the type of event was based on the assumption, that it is not necessary to introspect (or worse, try to deserialize a payload using Jackson)

Entirely true - but that is what has been implemented here (?) Only looking at the content-type, and if
- the content-type is a notification-type, and 
- if the helper class (HonoDemultiplexer) is used,

then (and only then)

- the notification object is built by using jackson.

If no helper class is used, the event is (as always) available in the eventHandler callback (no jackson involved).